### PR TITLE
feat(ingest): SageMaker jobs and models

### PR DESCRIPTION
**Pull request by kevinhu**
Adds support for SageMaker models and all eight job types. Also ingests all datasets fed into + generated by jobs as well as cases where one job spawns another (for instance, hyperparameter tuning jobs spawn multiple training jobs during the search process).

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by kevinhu** [comment]
(don't merge just yet – switching to representing each job with own flow + custom browse paths)

File: metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/jobs.py [hunk 1]
```diff
@@ -0,0 +1,1035 @@
...
+    describe_status_key: str
+    # job-specific mapping from boto3 status strings to DataHub-native enum
+    status_map: Dict[str, str]
+
+    # name of function for processing job for ingestion
+    processor: str
+
+
+# map from SageMaker job code to metadata on API access commands, fields, and processors
+SAGEMAKER_JOB_TYPES = {
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/jobs.py:48]: nice! this is a pretty clean solution, all things considered

File: metadata-ingestion/src/datahub/emitter/mce_builder.py [hunk 2]
```diff
@@ -73,6 +73,27 @@ def make_ml_feature_table_urn(platform: str, feature_table_name: str) -> str:
     )
 
 
+def make_ml_model_urn(platform: str, model_name: str, env: str) -> str:
+
+    return f"urn:li:mlModel:(urn:li:dataPlatform:{platform},{model_name},{env})"
+
+
+def make_s3_urn(s3_uri: str, env: str, suffix: Optional[str] = None) -> str:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/emitter/mce_builder.py:81]: seems odd to have this here - can we move it to one of the aws common files?

File: metadata-models/src/main/pegasus/com/linkedin/datajob/DataJobInfo.pdl [hunk 3]
```diff
@@ -35,7 +35,7 @@ record DataJobInfo includes CustomProperties, ExternalReference {
   /**
    * Datajob type
    */
-  type: union[AzkabanJobType]
```
Comment (hsheth2) [inline metadata-models/src/main/pegasus/com/linkedin/datajob/DataJobInfo.pdl:38]: should we add a note that `AzkabanJobType` is deprecated?

File: metadata-ingestion/examples/mce_files/bootstrap_mce.json [hunk 4]
```diff
@@ -1,1904 +1,1935 @@
...
-                            "firstName": null,
-                            "lastName": null,
-                            "fullName": {
-                                "string": "Data Hub"
-                            },
-                            "countryCode": null
-                        }
-                    }
-                ]
+  {
```
[the remark this answers is not in the document]
Comment (kevinhu) [reply metadata-ingestion/examples/mce_files/bootstrap_mce.json:2]: My bad – I added DataPlatform definitions for Feast and SageMaker – same as in metadata-ingestion/examples/mce_files/data_platforms.json.

File: metadata-ingestion/src/datahub/emitter/mce_builder.py (hunk 2 above)
Comment (kevinhu) [reply metadata-ingestion/src/datahub/emitter/mce_builder.py:81]: Yep, will do!

File: metadata-models/src/main/pegasus/com/linkedin/datajob/DataJobInfo.pdl (hunk 3 above)
Comment (kevinhu) [reply metadata-models/src/main/pegasus/com/linkedin/datajob/DataJobInfo.pdl:38]: Yes!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**